Add issue template form for suggesting new datasets

davanstrien opened the pull request:
Initial issue form template for proposing new datasets. I have made most fields optional -- happy for suggestions for any of these to be required. 

My proposed workflow is that people suggest a dataset through this form. They may then 'claim' this dataset and work on adding it or they might leave someone else to work on this aspect of it. 

In general, I have tried to ensure we capture the main information whilst not making the form too long. I did wonder if we should have some items related to the proposed way of adding to the hub i.e. script through directly sharing files but I think this might be better done in the issue discussion after the dataset is proposed?

davanstrien commented:
cc @albertvillanova

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+description: Suggest a new Library, Archive or Museum dataset to include in the Hugging Face Hub.
+title: "[Add Dataset]: "
+labels: ["dataset"]
+assignees: "davanstrien"
```
I would remove the assignee: people will self-assign an issue when willing to add the dataset themselves.

davanstrien replied:
that makes sense, I will remove this

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+    id: contact
+    attributes:
+      label: Contact Details
+      description: How can we get in touch with you if we need more info? (optional)
```
I'm thinking that if we need further details about the dataset adding proposal, we could use just the Issue page (the person who opened it will get notified).

Maybe we could use this field to ask contact information about the dataset custodian/owner: maybe we may need to contact the data custodian/owner to verify the license, to ask clarifications about the data structure,... We could ask for an email or their website (maybe there is a contact form there).

One option we should take into account is the possibility of making the dataset *gated*: some custodians allow access to their data only if we share with them the user email, so that they can be contacted in the future if some data item needs being removed. This is the case of Mozilla Common Voice corpus: see an example of gated dataset e.g. https://huggingface.co/datasets/mozilla-foundation/common_voice_9_0 
In this case, we should contact the data owners to check if they agree on having their data on the Hub under a gated dataset.

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+name: Add dataset
+description: Suggest a new Library, Archive or Museum dataset to include in the Hugging Face Hub.
+title: "[Add Dataset]: "
+labels: ["dataset"]
```
We have to agree on this step on the workflow: once a person proposes a dataset addition
- does it immediately gets into the project table and another person can start working on adding it immediately?
- or should it be first validated before getting into the project table?

Some kind of basic validations:
- the dataset is not duplicated
- its license certainly allows being added
- it is indeed a LAM-kind dataset
- ...

If we agree on the validation step, then I would propose:
- Create the issue with "dataset candidate" label
- Once validated, update the label with "dataset"

davanstrien replied:
I think it makes sense to have a validation step, I will document the things to check there and we can discuss how to manage that when we meet next. 

I will update the tag to "dataset candidate"

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+name: Add dataset
+description: Suggest a new Library, Archive or Museum dataset to include in the Hugging Face Hub.
+title: "[Add Dataset]: "
```
What about suggesting the the user proposes a dataset name?
```suggestion
title: "Add dataset: [dataset_name_in_snake_case]"
```

I would also propose to align with Datasets and naming it with snake_case (all in lowercase and word separated by underscores).
The validation step could also check this and rename the dataset if needed.

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+        - Other
+    validations:
+      required: true
+  - type: checkboxes
+    id: type-of-dataset
+    attributes:
+      label: Type of dataset
+      description: What type of datset is this?
+      options:
+        - label: A large unlabeled text corpus, i.e. something that could be used for training language models
+        - label: A large unlabeled image corpus, i.e. something that could be used for pre-training image models
+        - label: An annotated/labelled dataset, i.e. somethng that could be used to train/evaluate a machine learning model
+        - label: A benchmark dataset, i.e. a dataset that has been used to evaluate the performance of machine learning models
+        - label: Other
+    validations:
+      required: true
```
I'm not sure about the pertinence of this field.

In Datasets we have the concept of "tasks" and a dataset can be used for multiple tasks. The list is long: see https://github.com/huggingface/datasets/blob/master/src/datasets/utils/resources/tasks.json

davanstrien replied:
My main idea here was to try and ensure people don't add things that are out of scope for machine learning (e.g. very small collections of text without labels). However, since we will probably do an initial check to validate proposed datasets we can catch any issues with datasets there. 

I suggest we remove this from this section and people can align with the "tasks" when creating loading scripts etc. WDYT?

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+        - label: Other
+    validations:
+      required: true
+  - type: input
+    id: licence
+    attributes:
+      label: Dataset licence
+      description: Please include the license for the dataset here (either include the licence or a link) 
```
I'm thinking maybe it could be more useful having a dropdown, like we have when adding a dataset on the Hub: click inside the License field here https://huggingface.co/new-dataset

We have also a list in Datasets: https://github.com/huggingface/datasets/blob/master/src/datasets/utils/resources/licenses.json

Optionally, we could also have a textarea in case it has a specific/custom license: the user can put the text of the license or a link to it.

albertvillanova replied:
Maybe just proposing some of the most usual licenses? So that it will be faster to find it...

davanstrien replied:
My main reason for doing this was to avoid a long dropdown menu.  Perhaps a solution is to include the most common ones as a dropdown (the various creative commons licenses will likely capture >80% of datasets) and have an input box for ones with more complicated licenses/multiple licenses. 

I can add the most common ones as a dropdown choice if that sounds okay?

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,79 @@
+      label: How can you access this data
+      description: How can one access this data?
+      options:
+        - label: Via an open API
+        - label: Via an API that requires free registration 
+        - label: As a download from a repository/website 
```
What is the most usual way in your opinion?

If it is by a download, I would put this option first.

Why not using a dropdown here?

davanstrien replied:
I will switch this to a dropdown -- I think it's probably an even split but I will add the download one at the top since those are probably going to be the first ones people think to propose.

albertvillanova reviewed (commented):
Thanks a lot for this template: definitely useful if we would like to have the dataset metadata with a minimal structure.

davanstrien commented:
@albertvillanova  I think I've addressed your suggestions now. I also realised I hadn't included the most important field (a URL for the dataset) so I've added that now.

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+      - Creative Commons Attribution 2.0 Generic
+      - Creative Commons Attribution 2.5 Generic
+      - Creative Commons Attribution 3.0 Unported
+      - Creative Commons Attribution 3.0 Austria
+      - Creative Commons Attribution 3.0 United States
```
Are these common?

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+      - Creative Commons Attribution No Derivatives 4.0 International
+      - Creative Commons Attribution Share Alike 1.0 Generic
+      - Creative Commons Attribution Share Alike 2.0 Generic
+      - Creative Commons Attribution Share Alike 2.0 England and Wales
```
The same as above

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+      - Creative Commons Attribution Share Alike 2.0 England and Wales
+      - Creative Commons Attribution Share Alike 2.5 Generic
+      - Creative Commons Attribution Share Alike 3.0 Unported
+      - Creative Commons Attribution-Share Alike 3.0 Austria
```
The same

davanstrien replied:
I've reduced this to the most recent/international version of the Creative Commons licences.  Hopefully, that catches a good chunk of datasets. If we keep getting another licence under the other field we can add this to the dropdown.

albertvillanova replied:
Great!

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+    attributes:
+      value: |
+        Thank you for suggesting a dataset!
+  - type: input
+    id: data-url
+    attributes:
+      label: A URL for this dataset
+      description: A URL for information about this dataset i.e. a repository landing page 
+    validations:
+      required: true
```
Good catch!

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+      value: |
+        Thank you for suggesting a dataset!
+  - type: input
+    id: data-url
```
I would also propose to make the IDs as simple as possible: these can be passed as URL keyword arguments besides `template=add_dataset.yml` (so it's implicit they refer to adding a dataset)
```suggestion
    id: url
```

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+    validations:
+      required: true
+  - type: textarea
+    id: data-description
```
```suggestion
    id: description
```

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+    validations:
+      required: true
+  - type: dropdown
+    id:  dataset-modality
```
```suggestion
    id:  modality
```

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+    validations:
+      required: false
+  - type: dropdown
+    id: dataset-access
```
```suggestion
    id: accessibility
```

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+    validations:
+      required: true
+  - type: checkboxes
+    id: confirm-open
```
```suggestion
    id: open
```

albertvillanova commented on .github/ISSUE_TEMPLATE/add_dataset.yml:
```diff
@@ -0,0 +1,122 @@
+  - type: checkboxes
+    id: confirm-open
+    attributes:
+      label: Confirm the dataset is open
+      description: Please confirm the dataset is currently openly available 
```
Maybe to avoid ambiguity with the word "open"...
```suggestion
      label: Confirm the dataset has an open licence
      description: Please confirm the dataset is currently available under an open licence
```

albertvillanova reviewed (commented):
And last comments.

After this, I guess we can merge. If we think something should be added/removed, we can open a subsequent PR.

albertvillanova reviewed (approved):
Thanks @davanstrien, great job!

This template will be very useful: we can start adding dataset porposals.
